In [1]:
import pandas as pd

In [2]:
titanic_df = pd.read_csv("titanic_data.csv")
titanic_df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [3]:
titanic_df.shape

(891, 12)

In [4]:
titanic_df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [5]:
titanic_df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [6]:
titanic_df.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

In [7]:
titanic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [8]:
titanic_df[["Age","Fare"]].mean()

Age     29.699118
Fare    32.204208
dtype: float64

In [9]:
titanic_df[["Age","Fare"]].describe()

Age        Fare
count  714.000000  891.000000
mean    29.699118   32.204208
std     14.526497   49.693429
min      0.420000    0.000000
25%     20.125000    7.910400
50%     28.000000   14.454200
75%     38.000000   31.000000
max     80.000000  512.329200

**Description**: 
1. Average age of passengers is about 30, which suggest most of the passenger are young between 20 to 40.
2. Median fare is around 14.45, indicating that 50% of passengers paid less than this amount.
3. The fare range is wide (0 to 512), showing a large variation in ticket prices among passengers.
4. Age also shows variation, ranging from 0.42 to 80 years, with 75% of passengers being 38 years or younger.
5. The standard deviation of fare is high (~49.7), indicating some passengers paid much higher than average.





In [10]:
titanic_df.value_counts("Survived")

Survived
0    549
1    342
Name: count, dtype: int64

In [11]:
titanic_df.value_counts("Sex")

Sex
male      577
female    314
Name: count, dtype: int64

In [12]:
titanic_df.value_counts(["Sex", "Survived"], normalize= True)

Sex     Survived
male    0           0.525253
female  1           0.261504
male    1           0.122334
female  0           0.090909
Name: proportion, dtype: float64

Interestingly, death rate of male population is double the survival rate of female counterparts in total population.


In [13]:
titanic_df[titanic_df["Sex"] == "female"].value_counts("Survived",ascending=True,normalize=True)

Survived
0    0.257962
1    0.742038
Name: proportion, dtype: float64

Out of total female population, one/third part of female population survived while the remaining 25% did not survive.

In [14]:
titanic_df[titanic_df["Embarked"] == "S"].value_counts(["Sex", "Survived"])


Sex     Survived
male    0           364
female  1           140
male    1            77
female  0            63
Name: count, dtype: int64

Out of 644 people travelling from Southampton,  364 male population did not survive ,and 63 female did not survive where as 140 female and 77 male survived.

In [15]:
titanic_df["Embarked"].unique()

array(['S', 'C', 'Q', nan], dtype=object)

In [16]:
titanic_df["Embarked"].nunique()

3

In [17]:
titanic_df.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

In [18]:
group_by_gender = titanic_df[["Sex","Fare","Age","Survived"]].groupby(by = "Sex")

In [19]:
group_by_gender.mean()

Fare        Age  Survived
Sex                                   
female  44.479818  27.915709  0.742038
male    25.523893  30.726645  0.188908

In [20]:
group_by_gender.value_counts(["Survived"])

Sex     Survived
female  1           233
        0            81
male    0           468
        1           109
Name: count, dtype: int64

In [25]:
group_by_embarked = titanic_df.groupby(by=["Embarked", "Survived"])

In [26]:
group_by_embarked.median(numeric_only=True)

PassengerId  Pclass    Age  SibSp  Parch      Fare
Embarked Survived                                                    
C        0               378.0     3.0  30.00    0.0    0.0  14.45830
         1               485.0     1.0  27.00    0.0    0.0  56.92920
Q        0               503.0     3.0  30.75    0.0    0.0   7.75000
         1               301.5     3.0  20.50    0.0    0.0   7.80835
S        0               455.0     3.0  28.00    0.0    0.0  10.50000
         1               445.0     2.0  28.00    0.0    0.0  26.00000

In [27]:
titanic_df[["Age","Fare"]].cov()

Age         Fare
Age   211.019125    73.849030
Fare   73.849030  2469.436846

In [28]:
titanic_df[["Age","Fare"]].corr()

Age      Fare
Age   1.000000  0.096067
Fare  0.096067  1.000000

In [33]:
sampled=titanic_df.sample(n=100,random_state=99)

In [34]:
sampled[["Age","Fare"]].corr()

Age      Fare
Age   1.000000  0.120865
Fare  0.120865  1.000000

In [36]:
pd.crosstab(
    titanic_df.Pclass,
    titanic_df.Survived,
    margins=True
)

Survived    0    1  All
Pclass                 
1          80  136  216
2          97   87  184
3         372  119  491
All       549  342  891

In [37]:
pd.crosstab([titanic_df.Embarked,titanic_df.Pclass],[titanic_df.Sex,titanic_df.Survived],margins=True)

Sex             female      male       All
Survived             0    1    0    1     
Embarked Pclass                           
C        1           1   42   25   17   85
         2           0    7    8    2   17
         3           8   15   33   10   66
Q        1           0    1    1    0    2
         2           0    2    1    0    3
         3           9   24   36    3   72
S        1           2   46   51   28  127
         2           6   61   82   15  164
         3          55   33  231   34  353
All                 81  231  468  109  889